File: sfn/error/SfnErrorHandler.cs
using sfn_ut.sfn.error;

namespace sfn_ut.sfn;

public abstract class SfnErrorHandler<PAYLOAD> : SfnStep {
  public abstract SfnErrorHandlerInput<PAYLOAD> Execute(SfnErrorHandlerInput<PAYLOAD> payload);

  public SfnErrorHandler() : base(StepType.Error) {
    // Do nothing
  }
}

File: java-sfn-unit-test/src/main/java/com/sfnut/sfn/SfnErrorHandler.java
package com.sfnut.sfn;

import com.sfnut.sfn.error.SfnErrorHandlerInput;

public abstract class SfnErrorHandler<PAYLOAD> extends SfnStep {
    
    public SfnErrorHandler() {
        super(StepType.ERROR);
    }

    public abstract SfnErrorHandlerInput<PAYLOAD> execute(SfnErrorHandlerInput<PAYLOAD> payload);
}
